### PR TITLE
Create people grid component and person datasource

### PR DESCRIPTION
**Pull request by ericsanner**
https://trello.com/c/m9B3AaET/31-create-featured-speaker-component

image

image

## Description / Motivation

1. Added a Person Datasource template
2. Added a People Grid datasource template and rendering

This component is flexible and can suit multiple needs on the site.  People can later be linked to sessions (which can be linked to events).  The people grid component has rendering parameters that allow for

1. Number of columns
2. Background color (paintbucket styles)
3. Optional link to person's bio
4. Optional links to social urls

## How Has This Been Tested?

1. Mobile, tablet and desktop
2. With and without the background style
3. With and without each of the content fields
4. With and without the link to social sites
5. Different numbers of columns

The link to biography was not tested, as that will come in a future story.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read the Contributing guide.
- [x] My code/comments/docs fully adhere to the Code of Conduct.
- [x] My change is a code change.
- [ ] My change is a documentation change and there are NO other updates required.

### CODE REVIEW

File: src/Project/Sugcon2024/items/Templates/Sugcon2024/Basic Components/Person/Content/SocialLink1.yml
Comment (lovesitecore) [inline src/Project/Sugcon2024/items/Templates/Sugcon2024/Basic Components/Person/Content/SocialLink1.yml:1]: This should be used with dedicated Social Network. Please Name LinkedIn or Twitter or so. We might associate this with Icons. And they should be always displayed in order.

**Review by lovesitecore — CHANGES_REQUESTED** [review]
The Items are not consitantly using the same module. You started using the Basic Components Module. I do see a point putting it there when this becomes something more generic. But the display of other person related data is different and also business wise different. So I personally would separate these out. Just have a Speaker Template and a Featured Speaker Component. You could inherit fields to the Speaker Template from a Person Base Template. But this could be a bit overengineered. We can discuss this in todays call.

File: src/Project/Sugcon2024/items/PlaceholderSettings/Sugcon2024/List Components.yml
[the remark this answers is not in the document]
Comment (ericsanner) [reply src/Project/Sugcon2024/items/PlaceholderSettings/Sugcon2024/List Components.yml:1]: As discussed on standup, this component can handle all of the uses cases for speakers, featured speakers and organizers.

File: src/Project/Sugcon2024/items/Templates/Sugcon2024/Basic Components/Person/Content/SocialLink1.yml
Comment (ericsanner) [reply src/Project/Sugcon2024/items/Templates/Sugcon2024/Basic Components/Person/Content/SocialLink1.yml:1]: I have renamed the social links to linkedin and twitter.  Should the twitter link be called "X"?

**Review by lovesitecore — APPROVED** [review]

**Pull request merged**